### ISSUE TITLE
TestingGateway checks static configuration rather than gateway configuration

### ISSUE THREAD

**Issue by gregbeech** [braintree/braintree_ruby#95]
When you create a `Braintree::Gateway` with per-instance config (i.e. do not use the static accessors on `Braintree::Configuration` then it may not be possible to use the `testing` gateway on it because [it looks at the static configuration rather than the gateway configuration](https://github.com/braintree/braintree_ruby/blob/master/lib/braintree/testing_gateway.rb#L40).

This line should really be:

~~~ruby
raise TestOperationPerformedInProduction if @config.environment == :production
~~~

**Comment by wjr1985** [comment]
Hi @gregbeech, thanks for pointing this out! We have this on our roadmap to make the change to the client library.

**Comment by raymondberg** [comment]
Thanks again, @gregbeech. This has been released as part of 2.48.0. 

https://github.com/braintree/braintree_ruby/commit/05d7b6527b0f359874940b547f50ae422b7804ca#diff-344b3d13077e86851c6d7c8518d98717R40
